redisinsight/ui/src/components/base/link/link.styles.ts

```diff
@@ -0,0 +1,74 @@
+import styled, { css } from 'styled-components'
+import { Link as RedisUiLink, LinkProps } from '@redis-ui/components'
+import { useTheme } from '@redis-ui/styles'
+
+// TODO [DA]: Export the color functionality and use both for Link and Text
+export type EuiColorNames =
+  | 'inherit'
+  | 'default'
+  | 'primary'
+  | 'text'
+  | 'subdued'
+  | 'danger'
+  | 'ghost'
+  | 'accent'
+  | 'warning'
+  | 'success'
+
+export type ColorType = LinkProps['color'] | EuiColorNames | (string & {})
+
+export type RiLinkProps = Omit<LinkProps, 'color'> & {
+  color?: ColorType
+}
+
+export interface MapProps extends RiLinkProps {
+  $color?: ColorType
+}
+
+export const useColorTextStyles = ({ $color }: MapProps = {}) => {
+  const theme = useTheme()
+  const colors = theme.semantic.color
+
+  const getColorValue = (color?: ColorType) => {
+    if (!color) {
+      return colors.text.primary500
+    }
+    switch (color) {
+      case 'inherit':
+        return 'inherit'
+      case 'default':
+      case 'primary':
+        return colors.text.primary500
+      case 'text':
+        return colors.text.neutral700
+      case 'subdued':
+        return colors.text.informative400
+      case 'danger':
+        return colors.text.danger600
+      case 'ghost':
+        return colors.text.neutral600
+      case 'accent':
+        return colors.text.notice600
+      case 'warning':
+        return colors.text.attention600
+      case 'success':
+        return colors.text.success600
+      default:
+        return color // any supported color value e.g #fff
+    }
```

Comment:
Nit: I like the map syntax for such features:

```ts
const colorMap: Record<string, string> = {
    inherit: 'inherit',
    default: colors.text.primary500,
    primary: colors.text.primary500,
    text: colors.text.neutral700,
    subdued: colors.text.informative400,
    danger: colors.text.danger600,
    ghost: colors.text.neutral600,
    accent: colors.text.notice600,
    warning: colors.text.attention600,
    success: colors.text.success600,
  }

const getColorValue = (color?: ColorType) => {
  if (!color) {
    return colors.text.primary500
  }

  return colorMap[color] ?? color
}
```

IMO its less verbose. We can define it outside `useColorTextStyles` as well.